### ISSUE TITLE
Enable and fix sign-conversion warnings

### ISSUE THREAD

**Issue by armandas** [bblanchon/ArduinoJson#1715]
We're using `-Wsign-conversion` GCC option and we're getting errors from ArduinoJson. The errors are caused by the following template code in `RamString.hpp`:

```c++
template <int N>
struct IsString<char[N]> : true_type {};

template <int N>
struct IsString<const char[N]> : true_type {};

template <int N>
inline SizedRamString adaptString(char s[N]) {
  return SizedRamString(s, strlen(s));
}
```

All other places use `template <size_t N>`.

Would it be possible to add `-Wsign-conversion` to the compiler options and fix the reported issues? I could create a PR.

**Comment by armandas** [comment]
@bblanchon Thanks for the quick update!

**Comment by bblanchon** [comment]
You're welcome.
Thank you very much for reporting this issue.

**Comment by armandas** [comment]
@bblanchon I checked again and I'm still getting a warning [here](https://github.com/bblanchon/ArduinoJson/blob/421ecec0ddae1a7195b3be17367fdbeb2de31bd4/src/ArduinoJson/Variant/VariantImpl.hpp#L171).

```ArduinoJson/src/ArduinoJson/Variant/VariantImpl.hpp: In member function 'bool ArduinoJson6192_71::VariantRef::set(char) const':
ArduinoJson/src/ArduinoJson/Variant/VariantImpl.hpp:171:27: error: conversion to 'signed char' from 'char' may change the sign of the result [-Werror=sign-conversion]
  171 |   return set<signed char>(value);
      |                           ^~~~~
```

I'm not really sure why it doesn't show up in CI, but here is the [MCVE on Compiler Explorer](https://godbolt.org/z/T9ba6n4z8).

**Comment by bblanchon** [comment]
From what I could see in Compiler Explorer, the error occurs only on ARM GCC.
To get this error with other GCC variants, we must pass `-funsigned-char`.
I'll try to add this flag to one of the builds to trigger an error in the CI.

**Comment by bblanchon** [comment]
The fix was published in [ArduinoJson 6.19.3](https://github.com/bblanchon/ArduinoJson/releases/tag/v6.19.3).
